src/Microsoft.DotNet.Build.Tasks/PackageFiles/Symbols.targets

```diff
@@ -224,4 +226,174 @@
     </ItemGroup>
   </Target>
 
+  <!-- This will be overridden if we're building with MicroBuild. -->
+  <Target Name="SignFiles">
+    <Message Text="Fake sign target.  Would sign: @(FilesToSign)" />
+  </Target>
+
+  <!--
+    Entry point: inject signed symbol catalogs into all specified symbol packages.
+  -->
+  <Target Name="InjectSignedSymbolCatalogIntoSymbolPackages"
+          DependsOnTargets="GenerateSymbolCatalogs;
+                            SignFiles">
+    <MSBuild Targets="InjectSignedSymbolCatalogsForPackage"
+             Projects="$(MSBuildProjectFile)"
+             Properties="SymbolPackageFilePath=%(SymbolPackageFileWithIndex.Identity);
+                         SymbolPackageIndex=%(SymbolPackageFileWithIndex.Index);
+                         SymbolCatalogIntermediateDir=$(SymbolCatalogIntermediateDir);"
+             Condition="'@(SymbolPackageFileWithIndex)'!=''" />
+  </Target>
+
+  <!--
+    Generate symbol catalogs for all symbol packages specified. Creates FilesToSign items for each
+    catalog so they can be sigend.
+  -->
+  <Target Name="GenerateSymbolCatalogs">
+    <PropertyGroup>
+      <SymbolCatalogIntermediateDir Condition="'$(SymbolCatalogIntermediateDir)' == ''">$(BaseIntermediateOutputPath)SymbolsCatalog\</SymbolCatalogIntermediateDir>
+      <ExtensionsToCatalog Condition="'$(ExtensionsToCatalog)'==''">.dll;.pdb;.a;.so;.dbg;.dylib;.dwarf</ExtensionsToCatalog>
+      <MakeCatCommand Condition="'$(MakeCatCommand)'==''">makecat -v</MakeCatCommand>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <SymbolPackageFile Include="$(SymbolPackagesToPublishGlob)" />
+    </ItemGroup>
+
+    <!-- Extract to the index of the symbol package, not file name, to avoid exceeding max path. -->
+    <AddItemIndices Input="@(SymbolPackageFile)">
+      <Output TaskParameter="Output" ItemName="SymbolPackageFileWithIndex" />
+    </AddItemIndices>
+
+    <RemoveDir Directories="$(SymbolCatalogIntermediateDir)"
+               Condition="Exists('$(SymbolCatalogIntermediateDir)')" />
+
+    <MSBuild Targets="GenerateSymbolCatalogsForPackage"
+             Projects="$(MSBuildProjectFile)"
+             Properties="SymbolPackageFilePath=%(SymbolPackageFileWithIndex.Identity);
+                         SymbolPackageIndex=%(SymbolPackageFileWithIndex.Index);
+                         ExtensionsToCatalog=$(ExtensionsToCatalog);
+                         SymbolCatalogIntermediateDir=$(SymbolCatalogIntermediateDir);
+                         MakeCatCommand=$(MakeCatCommand)" />
+
+    <!-- Find the catalogs to sign them. -->
+    <ItemGroup>
+      <FilesToSign Include="$(SymbolCatalogIntermediateDir)**\*.cat">
+        <Authenticode>$(InternalCertificateId)</Authenticode>
```

Comment:
Is `InternalCertificateId` the variable that already exists and is used for signing regular binaries?  If so, I think we'd be better off using a different variable like `SymbolCatalogCertificateId` in case a repo wants to sign binaries with a different certificate (or the file types a certificate supports changes).  We could default the new variable to `$(InternalCertificateId)` for now in the build definition.